arbor/mechcat.cpp

```diff
@@ -74,6 +79,33 @@ void mechanism_catalogue::derive(const std::string& name, const std::string& par
         info->globals.at(param).default_value = value;
     }
 
+    for (const auto& kv: ion_remap_vec) {
+        if (!info->ions.count(kv.first)) {
+            throw invalid_ion_remap(name, kv.first, kv.second);
+        }
+    }
+
+    std::unordered_map<std::string, ion_dependency> new_ions;
+    for (const auto& kv: info->ions) {
```

Comment:
Loops like this would benefit with some comments that explain what they aim to acheive. It isn't immediately obvious, particularly if the variables (e.g. `ion_remap_map` are not documented either).

Reply:
While `ion_remap_map` I thought was pretty clear from it's declaration and initialization, I'll add a purpose-of-this-code comment to this for loop.